src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -972,6 +972,39 @@ public async Task Alpn_H3_Success()
             await connection.DisposeAsync();
         }
 
+        [Fact]
+        public async Task H3_AltUsedHeader()
+        {
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer();
+
+            Http3LoopbackConnection connection = null;
+            HttpRequestData requestData = null;
+            Task serverTask = Task.Run(async () =>
+            {
+                connection = (Http3LoopbackConnection)await server.EstablishGenericConnectionAsync();
+                requestData = await connection.ReadRequestDataAsync(readBody: false);
+                Assert.NotNull(connection);
+                Assert.Throws<System.Exception>(() => requestData.GetSingleHeaderValue("alt-used"));
```

Comment:
If this assert fails, what will you see? Will you be able to determine what went wrong?
I'm not saying this is wrong or insufficient, or even that we need to change it in case it's sub-optimal. It's just a thing to think about as well when writing these asserts.